src/esphomelib/output/my9231_output_component.cpp

```diff
@@ -0,0 +1,189 @@
+// Based on:
+//   - https://github.com/xoseperez/my92xx
+
+#include "esphomelib/defines.h"
+
+#ifdef USE_MY9231_OUTPUT
+
+#include "esphomelib/log.h"
+#include "esphomelib/output/my9231_output_component.h"
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace output {
+
+static const char *TAG = "output.my9231";
+
+// One-shot select (frame cycle repeat mode / frame cycle One-shot mode)
+static const uint8_t MY9231_CMD_ONE_SHOT_DISABLE = 0x0 << 6;
+static const uint8_t MY9231_CMD_ONE_SHOT_ENFORCE = 0x1 << 6;
+// Reaction time of Iout
+static const uint8_t MY9231_CMD_REACTION_FAST = 0x0 << 5;
+static const uint8_t MY9231_CMD_REACTION_SLOW = 0x1 << 5;
+// Grayscale resolution select
+static const uint8_t MY9231_CMD_BIT_WIDTH_16 = 0x0 << 3;
+static const uint8_t MY9231_CMD_BIT_WIDTH_14 = 0x1 << 3;
+static const uint8_t MY9231_CMD_BIT_WIDTH_12 = 0x2 << 3;
+static const uint8_t MY9231_CMD_BIT_WIDTH_8 = 0x3 << 3;
+// Internal oscillator freq. select (divider)
+static const uint8_t MY9231_CMD_FREQUENCY_DIVIDE_1 = 0x0 << 1;
+static const uint8_t MY9231_CMD_FREQUENCY_DIVIDE_4 = 0x1 << 1;
+static const uint8_t MY9231_CMD_FREQUENCY_DIVIDE_16 = 0x2 << 1;
+static const uint8_t MY9231_CMD_FREQUENCY_DIVIDE_64 = 0x3 << 1;
+// Output waveform
+static const uint8_t MY9231_CMD_SCATTER_APDM = 0x0 << 0;
+static const uint8_t MY9231_CMD_SCATTER_PWM = 0x1 << 0;
+
+MY9231OutputComponent::MY9231OutputComponent(const GPIOOutputPin &pin_di,
+                                             const GPIOOutputPin &pin_dcki,
+                                             uint16_t num_channels,
+                                             uint8_t num_chips,
+                                             uint8_t bit_depth, bool update)
+    : pin_di_(pin_di),
+      pin_dcki_(pin_dcki),
+      num_channels_(num_channels),
+      num_chips_(num_chips),
+      bit_depth_(bit_depth),
+      update_(update) {}
+
+void MY9231OutputComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MY9231OutputComponent.");
+  this->pin_di_.setup();
```

Comment:
```suggestion
  this->pin_di_->setup();
```

Same for other calls.